### PR TITLE
Add/fix ModelParallel layout map for Mixtral

### LINKED ISSUE #2833: Add ModelParallel layout map for MixtralBackbone
Mixtral has no layout map, so it can't be sharded with `ModelParallel`. This adds one along the Gemma/Mistral lines: the query kernel `(hidden, heads, head_dim)` shards hidden on data and heads on model, `reverse_embeddings` `(hidden, vocab)` shards vocab on model to match `embeddings`, and the GQA key/value kernels stay replicated on the model axis to avoid `IndivisibleError` when it doesn't divide `num_key_value_heads`. Mesh-shape and live-preset sweeps use width classes derived from the real `mixtral_8_7b` presets (4:1 GQA, 8 experts / top-2), the latter guarded by a per-width memory budget.

Part of #2807, added in #2817.

### PR DESCRIPTION
**Pull request by pctablet505**
Closes #2833

Part of https://github.com/keras-team/keras-hub/issues/2807. Depends on #2809 (`TestCase.run_distribution_test`) for CI to pass -- this PR's own diff is independent (based on master, not stacked), so it stays small and reviewable regardless of merge order.

## Summary

- Adds `MixtralBackbone.get_layout_map` for tensor-parallel sharding under `keras.distribution.ModelParallel`, mirroring the pattern already used by Gemma/Mistral.
- The query kernel and `reverse_embeddings` sharding axes follow the Megatron-style column-parallel convention used by the rest of the attention block:
  - Query kernel `(hidden, heads, head_dim)`: hidden (the contracting dim) is sharded on the data-parallel axis and heads on the model-parallel axis, consistent with `attention_output`'s rule.
  - `reverse_embeddings` `(hidden, vocab)`: data-parallel axis on the contracting hidden dim, model-parallel axis on vocab (mirrors the `embeddings` rule).
  - Key/value kernels are sized by `num_key_value_heads` (GQA) and are left fully replicated on the model axis to avoid `IndivisibleError` when the model-mesh dimension doesn't divide that (typically much smaller) axis, while hidden still shards on the data-parallel axis.
- Adds a mesh-shape sweep (`test_layout_map_mesh_shapes`, Tier 2) over two memory-scaled width classes derived from the real `mixtral_8_7b_en`/`mixtral_8_instruct_7b_en` preset ratios (4:1 GQA, 8 experts/top-2 MoE), across a capped set of mesh shapes.
- Adds a live-preset sweep (`test_layout_map_live_presets`, Tier 3) that fetches real preset configs and asserts sharding/coverage, with a per-width-class memory-budget guard so it never attempts a full-scale build on constrained hardware (subTest + logged skip reason instead of a silent drop or hard failure).

## Verification

Tests were run against a checkout that already has #2809's `TestCase.run_distribution_test` helper, since this branch's own base (master) doesn't yet include it -- this branch's diff itself is independent of #2809 and stays small regardless of merge order.

Backend: JAX, CPU, `XLA_FLAGS=--xla_force_host_platform_device_count=16` for the multi-device mesh-shape sweep.

```
$ KERAS_BACKEND=jax pytest keras_hub/src/models/mixtral/mixtral_backbone_test.py -k "not live_presets and not mesh_shapes" -v
3 passed, 3 skipped (test_session, test_saved_model, test_distribution [needs >1 device]) in 40.73s

$ XLA_FLAGS=--xla_force_host_platform_device_count=16 KERAS_BACKEND=jax pytest keras_hub/src/models/mixtral/mixtral_backbone_test.py -k "not live_presets" -v -n 2
16 passed, 2 skipped (test_session, test_saved_model) in 51.42s
  - test_distribution: PASSED
  - test_layout_map_mesh_shapes_mixtral8x7b-small_mesh_{2x4,1x8,4x4,2x2x2,1x1x8,2x2x4}: all PASSED
  - test_layout_map_mesh_shapes_mixtral8x7b-base_mesh_{2x4,1x8,4x4,2x2x2,1x1x8,2x2x4}: all PASSED
  - test_num_parameters, test_backbone_basics: PASSED

$ XLA_FLAGS=--xla_force_host_platform_device_count=16 KERAS_BACKEND=jax pytest keras_hub/src/models/mixtral/mixtral_backbone_test.py -k "live_presets" -v -s
1 skipped (kaggle_key_required; gated the same way in CI as the equivalent Gemma/Mistral tests)
```

All mesh shapes tested are capped at <=16 total simulated devices for memory reasons: `(2,4), (1,8), (4,4), (2,2,2), (1,1,8), (2,2,4)`.

## Limitations

The model-parallel mesh axis can't usefully exceed a preset's `num_query_heads`, since that's the axis actually being tensor-parallelized here -- see [the design doc's "hard limit" section](https://github.com/keras-team/keras-hub/issues/2807#issuecomment-5006606419) for why. For Mixtral's two current presets:

| preset | num_query_heads | safe model-axis up to | first failing model-axis |
|---|---|---|---|
| `mixtral_8_7b_en` | 32 | 32 | 64 |
| `mixtral_8_instruct_7b_en` | 32 | 32 | 64 |

Both presets use 4:1 GQA (`num_key_value_heads=8`) and top-2/8-expert MoE, but the failures at model-axis=64 are driven entirely by the query/attention-output kernels (i.e. the query-head ceiling above) -- neither preset shows a separate MoE-expert-dim or hidden/intermediate-dim failure at a different threshold in the swept range.

### CODE REVIEW

**Comment by pctablet505** [comment]
@gemini-code-assist review

**Comment by pctablet505** [comment]
@gemini-code-assist review
